SYCL/Plugin/interop-experimental-single-TU-SYCL-CUDA-compilation.cpp

```diff
@@ -0,0 +1,207 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %cuda_options -x cuda %s -lcudart -lcuda -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// REQUIRES: cuda && cuda_dev_kit
+
+#include <cuda.h>
+#include <sycl/sycl.hpp>
+
+// ------------------------------------------------------------------------- //
+
+// device-sum: test_cuda_function_0 = X
+// host-sum: test_cuda_function_0 = -X
+template <typename T> __device__ T test_cuda_function_0(T a, T b) {
+  return a + b;
+}
+template <typename T> __host__ T test_cuda_function_0(T a, T b) {
+  return -a - b;
+}
+
+// device-sum: test_cuda_function_2 + test_cuda_function_3 = 0
+// host-sum: test_cuda_function_3 = 0
+__device__ inline float test_cuda_function_2(float a, float b) {
+  return -sin(a) + b;
+}
+__device__ inline float test_cuda_function_3(float a, float b) {
+  return sin(a) - b;
+}
+__host__ inline float test_cuda_function_3(float a, float b) { return 0; }
+
+// device/host-sum: test_cuda_function_4 = 0
+__device__ __host__ inline float test_cuda_function_4(float a, float b) {
+  return (a - b) + (b - a);
+}
+
+// device-sum: test_cuda_function_5 + test_cuda_function_6 = X
+// host-sum: test_cuda_function_5 + test_cuda_function_6 = -X
```

Comment:
```suggestion
// device-sum: test_cuda_function_5 + test_cuda_function_6 = 0
// host-sum: test_cuda_function_5 + test_cuda_function_6 = 0
```

or am I missing anything?

Reply:
Sure, true.